packages/kbn-test/src/auth/saml_auth.ts

```diff
@@ -25,10 +25,12 @@ export class Session {
   readonly cookie;
   readonly email;
   readonly fullname;
-  constructor(cookie: Cookie, email: string, fullname: string) {
+  readonly username;
+  constructor(cookie: Cookie, email: string, fullname: string, username: string) {
     this.cookie = cookie;
     this.email = email;
     this.fullname = fullname;
+    this.username = username;
```

Comment:
I'm mostly concerned about consistency b/w Cloud SAML and mockIdp we use for CI/local runs. Did you try to run any of the tests against real MKI project?

Reply:
[QAF](https://docs.google.com/document/d/1tiax7xoDYwFXYZjRTgVKkVMjN-SQzBWk4yn1JY6Z5UY/edit?pli=1#heading=h.f6esfmqe4nh5) should make testing it against MKI, from your local very easy @machadoum

Reply:
Confirmed with Oleg/kibana-security team: `username` is **auto-generated and not controlled by the user**. I believe the right approach to retrieve it is to use Kibana API (already available with `getSecurityProfile` function)

Reply:
I exported the username because some tests assert that the generated documents have the fields `created_by` and `generated_by` with the correct username.

Let me try using `getSecurityProfile` instead of updating the Session.

Reply:
Wait... I need help understanding the problem. The username is already coming from `getSecurityProfile(..).username`

Reply:
Yes, username can and should be fetched via Kibana API, actually defining any user metadata through saml authentication is not the right approach. It is simply can be incorrect/out dated.

#186279 I moved user profile fetching to be done explicitly via `sessionManager.getUserData()`:

So if you need to check any of user profile data, you have to call Kibana API and rely on it as a source of truth.

Reply:
@machadoum can you please use the above method to unblock the PR?